specs/consensus.md

```diff
@@ -140,10 +142,19 @@ TODO define fraud proof for invalid block here
 
 Once parsed, the following checks must be `true`:
 
-1. The commitments of the [erasure-coded extended](./data_structures.md#2d-reed-solomon-encoding-scheme) `availableData` must match those in `header.availableDataHeader`.
+1. The commitments of the [erasure-coded extended](./data_structures.md#2d-reed-solomon-encoding-scheme) `availableData` must match those in `header.availableDataHeader`. Implicitly, this means that both rows and columns must be ordered lexicographically by namespace ID since they are committed to in a [Namespace Merkle Tree](data_structures.md#namespace-merkle-tree).
+1. Length of `availableData.intermediateStateRootData` == length of `availableData.transactionData` + length of `availableData.evidenceData`.
```

Comment:
Why would evidence add to the intermediate state roots?

Reply:
Ah OK, that's assuming the evidence has an immediate effect on the state root.

Reply:
Yep, which it should! Each evidence actually mutates the state, i.e. is a state transition, and so should have an associated intermediate state root.